DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/ip_authoring_mmhost/src/mmhost.cpp

```diff
@@ -0,0 +1,184 @@
+#include <iomanip>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/prototype/interfaces.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+using namespace sycl;
+using ext::intel::experimental::property::usm::buffer_location;
+
+template <typename T>
+inline std::shared_ptr<T> make_malloc_shared(queue &q, int n, int kBL) {
```

Comment:
I am not sure if we want to make this part of the tutorial. Bring this up as an open in the next team meeting please.